modules/effects/src/provide_effects.ts

```diff
@@ -0,0 +1,64 @@
+import { ENVIRONMENT_INITIALIZER, inject, Provider, Type } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { EffectsRunner } from './effects_runner';
+import { EffectSources } from './effect_sources';
+import { rootEffectsInit as effectsInit } from './effects_actions';
+
+/**
+ * Runs provided effects. Can be called at the root or feature level.
+ * Unlike the `StoreModule.forRoot` method, this function does not need
+ * to be called at the root level if there are no root effects.
+ *
+ * @usageNotes
+ *
+ * ### Providing root effects
+ *
+ * ```ts
+ * bootstrapApplication(AppComponent, {
+ *   providers: [provideEffects([RouterEffects])],
+ * });
+ * ```
+ *
+ * ### Providing feature effects
+ *
+ * ```ts
+ * const booksRoutes: Route[] = [
+ *   {
+ *     path: '',
+ *     providers: [provideEffects([BooksApiEffects])],
+ *     children: [
+ *       { path: '', component: BookListComponent },
+ *       { path: ':id', component: BookDetailsComponent },
+ *     ],
+ *   },
+ * ];
+ * ```
+ */
+export function provideEffects(effects: Type<unknown>[]): Provider[] {
+  return [
+    effects,
+    {
+      provide: ENVIRONMENT_INITIALIZER,
+      multi: true,
+      useValue: () => {
+        const effectsRunner = inject(EffectsRunner);
+        const effectSources = inject(EffectSources);
+        const shouldInitEffects = !effectsRunner.isStarted;
```

Comment:
We should use a provider token here that checks to see if there is a parent token above this injector. If not then start the effects and we don't have to use property on the class

Reply:
I'm not sure it would work when `provideEffects` is called multiple times within the context of the same injector.